core/src/main/scala/kafka/coordinator/DelayedJoin.scala

```diff
@@ -31,8 +31,9 @@ import kafka.server.DelayedOperation
  */
 private[coordinator] class DelayedJoin(coordinator: GroupCoordinator,
                                        group: GroupMetadata,
-                                       rebalanceTimeout: Long)
-  extends DelayedOperation(rebalanceTimeout) {
+                                       delayMs: Int,
+                                       val rebalanceTimeOut: Long)
+  extends DelayedOperation(delayMs) {
```

Comment:
Is using `delayMs` here actually correct? For `DelayedOperation` the timeout provided is supposed to be the maximum amount of time. The KIP says

> Change the GroupCoordinator so that upon receiving the initial JoinGroupRequest for a new or empty group, it will change the group’s state to InitialRebalance and delay the operation by min(rebalanceTimeout, group.initial.rebalance.delay.ms).

That seems to suggest there should be some combination of `delayMs` and `rebalanceTimeOut` used here. Neither one is guaranteed to be the largest or smallest of the two.